manipulation.go

```diff
@@ -334,8 +370,23 @@ func (s *Selection) WrapSelection(sel *Selection) *Selection {
 // most child of the given HTML.
 //
 // It returns the original set of elements.
-func (s *Selection) WrapHtml(html string) *Selection {
-	return s.wrapNodes(parseHtml(html)...)
+func (s *Selection) WrapHtml(htmlStr string) *Selection {
+	nodesMap := make(map[html.NodeType][]*html.Node)
+	var parent *html.Node
+	for _, context := range s.Nodes {
+		if context.Parent != nil {
+			parent = context.Parent
+		} else {
+			parent = &html.Node{Type: html.ElementNode}
+		}
+		nodes, found := nodesMap[parent.Type]
```

Comment:
The map (cache) should use the node name logic as in `eachNodeHtml` as key, not the Type.

Reply:
I agree, must have totally skipped over those. I'll add the same logic as in `eachNodeHtml`